lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -26,47 +27,71 @@ namespace esi {
 using namespace mlir;
 using namespace circt::esi;
 
-namespace {
+/// Figure out the number of bits a type takes on the wire in verilog. Doing it
+/// here is probably a giant hack. TODO: Establish a canonical method to get
```

Comment:
Any thoughts on this? I'm thinking a function in the `circt::sv` namespace is sufficient.

Reply:
I think it would make sense to have a global function in RTL/Types.h.  It's not clear what this will mean with extensible types though.

Reply:
The RTL dialect, not the SV dialect? I was thinking SV since the number of bits is technically a SystemVerilog thing (other dialects which lower from RTLL could represent types differently). Putting it in the SV dialect _could_ solve the extensibility issue since it is reasonable for the SV dialect to have a closed type system.

Reply:
I'm gonna leave this as-is while we work this out.

Reply:
https://llvm.discourse.group/t/getting-the-size-of-a-type-in-systemverilog/2414